### ISSUE TITLE
Problem with LA 104 firmware via USB

### ISSUE THREAD

**Issue by gumalexgum** [gabonator/LA104#88]
[](https://translate.google.com/website?sl=auto&tl=ru&hl=ru&anno=2&u=https://user-images.githubusercontent.com/101945675/159135237-53a87d3e-10fe-4271-a281-e4d23a0ff767.png)[](https://translate.google.com/website?sl=auto&tl=ru&hl=ru&anno=2&u=https://user-images.githubusercontent.com/101945675/159135137-d4483dd1-3e3b-4b1b-b1f5-7db27d174263.png)Hello! I really liked your product, but I can't seem to get it set up correctly. Help me to understand. I am not very strong in STM 32 and have not worked with them. It would be ideal to make a video tutorial on how to flash LA 104 in Windows. I'm using Windows 7 x 64 operating system and VirtualBox with Unix 64. At first I was able to write the sistem .hex file, but then the connection was lost while transferring files and the screen shows: "Uknown section name !" "Assertion failed in .. /sourse/main/Execue.cpp [355] : sectionType > 0, Application halted. Virtual disk 2 Mb starts with first button pressed. But file transfer often fails.

![image](https://user-images.githubusercontent.com/101945675/159136161-d35c8078-4eb8-4593-a7a8-5bd89c43c028.png)

![image](https://user-images.githubusercontent.com/101945675/159136224-94112f20-547b-4f18-808c-dad9281b9005.png)

**Comment by gumalexgum** [comment]
> @gumalexgum this thread is related only to webusb connection issues, please start a new issue. It seems that the shell.elf file is corrupted. Start the device with 4th button pressed (or 3rd, I can't remember exactly) to enter safe mode. Delete everything from the usb drive and copy the shell.elf again.

**Comment by gumalexgum** [comment]
Safe mode.  Windows: USB device not recognized!

**Comment by gumalexgum** [comment]
> @gumalexgum It seems that the shell.elf file is corrupted. Start the device with 4th button pressed (or 3rd, I can't remember exactly) to enter safe mode. Delete everything from the usb drive and copy the shell.elf again.

Sorry! Does not work! The USB device is not recognized by Windows.

**Comment by gabonator** [comment]
Show me screenshot of device manager. If the drive is just not accessible, try formatting it

**Comment by gumalexgum** [comment]
> Show me screenshot of device manager. If the drive is just not accessible, try formatting it

![image](https://user-images.githubusercontent.com/101945675/159137607-09e220f2-97bb-4ba0-8fcf-ffac453cb069.png)

**Comment by gabonator** [comment]
That is weird, can you try attaching it to another computer?

**Comment by gumalexgum** [comment]
Yes, I tried connecting to a Windows 7 x 32 laptop. Same thing. In my opinion, there was a failure during the transfer of files and an incorrect file name. Because of this, the USB is not connected.

**Comment by gumalexgum** [comment]
When copying system_la104.hex, system_la104.err appears first, which I will rename to system_la104.hex, after which system_la104.rdy appears. I don't know if this file is correct or not?

**Comment by gabonator** [comment]
No, do not rename them. Just copy for the first time, wait util the device gets disconnected, then copy the file again.

**Comment by gumalexgum** [comment]
Can I ask you to write down the points that I need to do and in what order?

one) ...
2) ...
3) ...
...

**Comment by gumalexgum** [comment]
![file](https://user-images.githubusercontent.com/101945675/159524725-5b1b50ee-e3db-43ce-81ee-a8fe6287de39.jpg)

**Comment by gumalexgum** [comment]
What USB device should LA104 detect when connected to a computer? And why does Windows 7 not detect the device ID?

**Comment by mgazza** [comment]
people seeing this issue cd to the LA104 drive and run `. -type f -name '.*' -delete`

**Comment by pontusbredin** [comment]
Hi, I ran into problems, but solved it in the end. I hope this guide can help others.

1. Press and hold SMPL button and turn on the device.
2. Connect the device via USB to a Windows 11 computer.
3. Open the drive in File Explorer and make sure you can see hidden files.
4. Select all files and press delete to remove them.
5. Copy the latest system_la104.hex file to the drive.
6. After a while the drive will disappear and appear again. At this point there shold be a file called system_la104.err on the drive.
7. Copy system_la104.hex file to the drive again.
8. After a while the drive will disappear and appear again. At this point there shold be one more file called system_la104.rdy on the drive.
9. Remove the file system_la104.err.
10. Turn off the device and disconnect the USB cable.
11. Turn on the device and connect the USB cable again. Now Windows should prompt you to format the drive. On the device an error about shell.elf will appear, ignore this for now. If the drive has been formatted you will not get this prompt.
12. Format the drive with the file system FAT. Perform the format even if everything looks okey. Fast format will work just fine.
13. Now unzip la104apps.zip into a folder on your computer.
14. Enter the folder and select all the files and select copy.
15. Paste all the files in the newly formatted device drive. Unmount the USB drive. Disconnect the USB cable.
16. Power off and the on the device.
17. Done

If step 11-12 failes, do this:
1. Turn off the device.
2. Press and hold the ESC button while turning on the device. This enters safe mode.
3. A drive should now appear.
4. Follow steps 12-16 above. Don't skip formatting the drive!

You might corrupt the drive some times. In that case follow the last steps 1-4.

Thank you Gabriel Valky for all the awesome work you've put in to this project!!!
